pkg/apis/registryman/v1alpha1/types.go

```diff
@@ -261,6 +261,13 @@ type ProjectSpec struct {
 
 	// Scanner specifies the name of the assigned scanner.
 	Scanner string `json:"scanner,omitempty"`
+
+	// +kubebuilder:validation:Optional
+
+	// Trigger specifies the preferred replication trigger. If it is not
+	// possible to implement the selected replication trigger, the trigger
+	// may be overridden.
+	Trigger string `json:"trigger,omitempty"`
```

Comment:
I think that this would deserve some structured type as opposed to string, `cron */10 * * * *` value looks quite error prone.

Reply:
I was thinking a lot about this one. The more specific we are at the API level, the more difficult it is to implement a new registry provider. Our approach is to be permissive at the API and catch the errors during the semantic validation or execution phase.

Reply:
In that particular case, you convinced me anyway. I'll implement the structured type.